examples/lighting-app/nrfconnect/README.md

```diff
@@ -0,0 +1,290 @@
+[Nordic nRF52840 DK](https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF52840-DK)
+development kit, but is readily adaptable to other nRF52840-based hardware.
+
+The lighting example is intended to serve both as a means to explore the
+workings of CHIP, as well as a template for creating real products based on the
+Nordic platform.
+
+The example makes use of the CMake build system to generate the ninja build
+script. The build system takes care of invoking the CHIP library build with all
+necessary flags exported from the Zephyr environment.
+
+<a name="device-ui"></a>
+
+## Device UI
+
+The example application provides a simple UI that depicts the state of the
+device and offers basic user control. This UI is implemented via the
+general-purpose LEDs and buttons built in to the nRF52840 DK dev board.
+
+**LED #1** shows the overall state of the device and its connectivity. Four
+states are depicted:
+
+-   _Short Flash On (50ms on/950ms off)_ &mdash; The device is in an
+    unprovisioned (unpaired) state and is waiting for a commissioning
+    application to connect.
+
+*   _Rapid Even Flashing (100ms on/100ms off)_ &mdash; The device is in an
+    unprovisioned state and a commissioning application is connected via BLE.
+
+-   _Short Flash Off (950ms on/50ms off)_ &mdash; The device is full
+    provisioned, but does not yet have full network (Thread) or service
+    connectivity.
+
+*   _Solid On_ &mdash; The device is fully provisioned and has full network and
+    service connectivity.
+
+**Button #1** can be used to initiate a OTA software update as well as to reset
+the device to a default state.
+
+Pressing and holding Button #1 for 6 seconds initiates a factory reset. After an
+initial period of 3 seconds, all four LED will flash in unison to signal the
+pending reset. Holding the button past 6 seconds will cause the device to reset
+its persistent configuration and initiate a reboot. The reset action can be
+cancelled by releasing the button at any point before the 6 second limit.
+
+**LED #2** shows the state of the lighting.
+
+**Button #2** can be used to change the state of the lighting. This can be used
+to mimick a user manually switching the lighting. The button behaves as a
+toggle, swapping the state every time it is pressed.
+
+**Button #3** can be used to trigger a thread joiner. It should be use to
+commission to the thread network. Credentials will be presented in log messages.
+
+The remaining two LEDs (#3 and #4) and button #4 are unused.
+
+<a name="building"></a>
+
+## Building
+
+### Using Docker container
+
+The easiest way to get started with the example is to use nRF Connect SDK Docker
+image for CHIP applications. Run the following commands to start a Docker
+container:
+
+        $ mkdir ~/nrfconnect
+        $ mkdir ~/connectedhomeip
+        $ docker run --rm -it --privileged -v /dev:/dev -v ~/nrfconnect:/var/ncs -v ~/connectedhomeip:/var/chip nordicsemi/nrfconnect-chip
```

Comment:
I tried this yesterday and something the container did turned the /dev/null device on my host system into a regular file. I had to reboot to recover.

This is pretty risky - far worse than corrupting /dev/null could happen with root access to all devices. I'm not sure we should recommend this workflow.

Reply:
Interesting, it's never happened to me. I'll change the instruction to only expose serial devices.